pact/sale-contracts/conventional-auction/conventional-auction.pact

```diff
@@ -214,22 +235,28 @@
                 'bidder-guard:= previous-bidder-guard
               }
               (with-capability (REFUND_CAP sale-id)
-                (install-capability (coin.TRANSFER (escrow-account sale-id) previous-bidder prev-highest-bid))
-                (coin.transfer (escrow-account sale-id) previous-bidder prev-highest-bid)
+                (install-capability (fungible::TRANSFER (escrow-account sale-id) previous-bidder (fungible::get-balance (escrow-account sale-id))))
+                (fungible::transfer (escrow-account sale-id) previous-bidder (fungible::get-balance (escrow-account sale-id)))
               )
             )
             true
           )
 
-          ; Transfer amount from bidder to escrow-account
-          (coin.transfer-create bidder (escrow-account sale-id) (escrow-guard sale-id) bid)
+          ; Transfer amount (including fee if applicable) from bidder to escrow-account
+          (fungible::transfer-create bidder (escrow-account sale-id) (escrow-guard sale-id) (+ bid mk-fee))
 
           ; Write new bid and store highest bid in auction
           (write bids bid-id {
             "bidder": bidder
             ,"bidder-guard": bidder-guard
             ,"bid": bid
           })
+
+          ; Store marketplace fee configuration
+          (if (= mk-fee-percentage 0.0)
+            true
+            (write mk-fees bid-id mk-fee-spec)
```

Comment:
One liner check such as 
`(fungible::details (at 'mk-account mk-fee-spec))`
would be useful here, because if the account does not exist in the fungible, then winner might be blocked from processing the buy. We can also secure that that a random string does not get involved here.